File: Main/Globals/Classes/Decorators/PageRange.js
class PageRange
{
    #startPage;
    #endPage;

    constructor({startPage = 0, endPage = 0} = {})
    {
        this.setStartPage(startPage);
        this.setEndPage(endPage);
    }

    getStartPage()
    {
        return this.#startPage;
    }

    setStartPage(value)
    {
        if (value !== null)
        {
            value = parseInt(value, 10);
            if (isNaN(value))
            {
                value = 0;
            }
            else
            {
                value = Math.max(value, 0);
            }
        }
        this.#startPage = value;
    }

    getEndPage()
    {
        return this.#endPage;
    }

    setEndPage(value)
    {
        if (value !== null)
        {
            value = parseInt(value, 10);
            if (isNaN(value))
            {
                value = 0;
            }
            else
            {
                value = Math.max(value, 0);
            }
        }
        this.#endPage = value;
    }

    toJson()
    {
        return {
            startPage: this.getStartPage(),
            endPage: this.getEndPage(),
        };
    }

    static fromJson(json)
    {
        const instance = new PageRange({
            startPage: json.startPage ?? null,
            endPage: json.endPage ?? null
        });
        return instance;
    }
}

export default PageRange;

File: Agent/Globals/Classes/Decorators/PageRange.py

class PageRange:
    def __init__(self, start_page: int = 0, end_page: int = 0) -> None:
        self.set_start_page(start_page)
        self.set_end_page(end_page)

    def get_start_page(self) -> int:
        return self.__start_page

    def set_start_page(self, value: int) -> None:
        if value is not None:
            try:
                value = int(value)
                value = max(0, value)
            except (ValueError, TypeError):
                value = 0
        self.__start_page = value

    def get_end_page(self) -> int:
        return self.__end_page

    def set_end_page(self, value: int) -> None:
        if value is not None:
            try:
                value = int(value)
                value = max(0, value)
            except (ValueError, TypeError):
                value = 0
        self.__end_page = value

    def to_json(self) -> dict:
        return {
            'startPage': self.get_start_page(),
            'endPage': self.get_end_page(),
        }

    @classmethod
    def from_json(cls, data: dict) -> 'PageRange':
        instance = cls(
            start_page=data.get('startPage'),
            end_page=data.get('endPage')
        )
        return instance
